### PR TITLE
Calculate qos test's parameter in runtime

### PR DESCRIPTION
**Pull request by XuChen-MSFT**
### Description of PR

Summary:

Calculate qos test's parameter in runtime

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?

On brcm platfomr, qos testcase's parameter  is not static value.
have to calculate it dynamically.

#### How did you do it?

so far, only prepare invoking  of calculation utils.
so, just directly return qos param to testcase.

#### How did you verify/test it?

pass local test.

### CODE REVIEW

File: tests/qos/qos_sai_base.py [hunk 1]
```diff
@@ -881,6 +881,31 @@ def disablePacketAging(
...
+            bufferConfig['BUFFER_PG'] = json.loads(duthost.shell('sonic-cfggen -d --var-json "BUFFER_PG"')['stdout'])
+        except Exception as err:
+            logger.info(err)
+        return bufferConfig
+
+    def dutAsicConfig(self, asic, duthost):
+        asicConfig = {}
+        # Only support to get brcm asic info, so far
+        if 'broadcom' in asic.lower():
+            output = duthost.shell('bcmcmd "g THDI_BUFFER_CELL_LIMIT_SP"', module_ignore_errors=True)
```
Comment (StormLiangMS) [inline tests/qos/qos_sai_base.py:899]: Is this register available for all BRCM asics?

**Review by StormLiangMS — DISMISSED** [review]
approved with a question.

File: tests/qos/qos_sai_base.py (hunk 1 above)
Comment (XuChen-MSFT) [reply tests/qos/qos_sai_base.py]: on th/th2/td2/td3 platform, command 'bcmcmd "g THDI_BUFFER_CELL_LIMIT_SP"' is always woring.
And add "try except" for THDI_BUFFER_CELL_LIMIT_SP register is unavailable case.

**Review by ZhaohuiS — APPROVED** [review]
LGTM

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**